ansible/devutil/devices/ansible_hosts.py

```diff
@@ -502,16 +506,41 @@ def _check_results(self, caller_info, module_info, results, module_ignore_errors
         if isinstance(self, AnsibleHosts):
             hosts_str = json.dumps(self.hostnames)
         elif isinstance(self, AnsibleHost):
-            hosts_str = json.dumps(self.hostnames[0])
-            results = results.get(self.hostnames[0], {})
+            hosts_str = self.hostnames[0]
         else:
             raise TypeError("Unsupported type of object: {}".format(type(self)))
 
-        if isinstance(module_info, dict):
+        unreachable = False
+        failed = False
+        for _, module_results in results.items():
+            if isinstance(module_results, list):
+                for result in module_results:
+                    if "reachable" in result and not result["reachable"]:
+                        unreachable = True
+                    if "unreachable" in result and result["unreachable"]:
+                        unreachable = True
+                    if "failed" in result and result["failed"]:
+                        failed = True
```

Comment:
Same as line527-532. Extract a method such as `check_failure()->unreachable, failed:` may be better?